### PR TITLE
Emit 'deprecated' operation property in openapi3

### PR DESCRIPTION
**Pull request by leni-msft**
fixes https://github.com/microsoft/cadl/issues/979

### CODE REVIEW

File: packages/openapi3/test/openapi-output.test.ts [hunk 1]
```diff
@@ -160,6 +160,19 @@ describe("openapi3: operations", () => {
     strictEqual(getThing.parameters[1].schema.minimum, 1);
     strictEqual(getThing.parameters[1].schema.maximum, 10);
   });
+
+  it("deprecate operations with @deprecated", async () => {
+    const res = await openApiFor(
+      `
+      @deprecated("use something else")
+      @route("/")
```
Comment (timotheeguerin) [inline packages/openapi3/test/openapi-output.test.ts:168]: Would just remove the `@route` and `@get` decorators for the test. They don't add much value but make the test longer

**Review by timotheeguerin — APPROVED** [review]

File: packages/openapi3/src/openapi.ts [hunk 2]
```diff
@@ -473,6 +474,8 @@ function createOAPIEmitter(program: Program, options: ResolvedOpenAPI3EmitterOpt
     emitRequestBody(parameters, visibility);
     emitResponses(operation.responses);
 
+    isDeprecated(program, op) && (currentEndpoint.deprecated = true);
```
Comment (timotheeguerin) [inline packages/openapi3/src/openapi.ts:477]: Think using an `if` make this a bit more readable in this case

Comment (leni-msft) [reply packages/openapi3/src/openapi.ts:477]: sure, will do

File: packages/openapi3/test/openapi-output.test.ts (hunk 1 above)
Comment (leni-msft) [reply packages/openapi3/test/openapi-output.test.ts]: good point.

**Pull request merged**